torch/csrc/utils/device_lazy_init.cpp

```diff
@@ -0,0 +1,43 @@
+#include <torch/csrc/utils/device_lazy_init.h>
+
+#include <torch/csrc/Exceptions.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/object_ptr.h>
+#include <iostream>
+namespace torch::utils {
+namespace {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables,modernize-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
+bool is_initialized[at::COMPILE_TIME_MAX_DEVICE_TYPES] = {false};
```

Comment:
Please use std::array here, as the construct you wrote here will initialize only first element of the array, see https://godbolt.org
```suggestion
std::array<bool, at::COMPILE_TIME_MAX_DEVICE_TYPES>  is_initialized{};
```

Reply:
Thanks for your suggestion. Done.